### PR TITLE
General t4/write frontend unit tests

### PR DESCRIPTION
**Pull request by sehersarin**
**Scrum Board Ticket:** General T4: Setup frontend unit tests using Jest
**Story:** General Codebase Tasks

**Description:** Scope involves updating file coverage for all current components to be 100% or as maximized as possible.
- Wrote unit tests for all frontend files that were created in previous sprints but no tests were written [Previous test report looked as follows prior to this story being developed.]

Screen Shot 2020-11-25 at 10 18 54 AM

**Testing Criteria (how I tested the code):** 
1. Ran the automated tests in the react-ui folder and ensured that there was a high file coverage and all tests passed.

**How Reviewers can Test Code:**
1. Run the code locally and ensure no complications when starting up.
    a.) Navigate inside the `react-ui` and `server` folders and run npm start.
2. Inside the `react-ui` folder, run` npm test`. All tests should pass and the test report should look like the screenshot attached.
    a) Tests written for this ticket include `index.js`, `Signup.js`, `WorkerInputPage.js`, `Profile.js`, `Home.js`, `404.js`, `Dashboard.js` and `Header.js`.

Screen Shot 2020-12-02 at 9 21 10 AM 

*Please note: Any tests that still haven't been updated (in red) are in scope for another PR (teammates developing frontend code for this sprint will be writing their own frontend tests for those tickets.)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,6 +5,19 @@ The overall problem the team aims to address is the non-confidentiality and inef
 
 The link to access the system on a web browser: https://connectme-msci342.herokuapp.com/ 
 
+### Implemented Functionality
```
Comment (sehersarin) [inline README.md:8]: Updated readme with tasks already implemented by the team over Sprints 1-4.

File: react-ui/src/components/Header.scss [hunk 2]
```diff
@@ -5,7 +5,7 @@
   }
   
   .App-header-blue {
-    background-color: #eae8ff;
+    background-color: #b8b1fd;
```
Comment (sehersarin) [inline react-ui/src/components/Header.scss:8]: Changed header colour to match new colour theme decided by team.

File: react-ui/src/components/LogInForm.test.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ import { Redirect, Route, withRouter, Link } from "react-router-dom";
 jest.mock('axios');
 
 
-describe('SignupForm component', () => {
+describe('LoginForm component', () => {
```
Comment (sehersarin) [inline react-ui/src/components/LogInForm.test.js:17]: Updated test name to keep consistent with file being tested.

File: react-ui/src/components/Header.js [hunk 4]
```diff
@@ -6,15 +6,9 @@ export default class Header extends Component {
     render() {
 
       let header;
-      if(this.props.color==="red"){
```
Comment (sehersarin) [inline react-ui/src/components/Header.js:9]: Removed these lines from header file as they are no longer used in our application.

File: react-ui/src/components/Header.scss (hunk 2 above)
Comment (prdbondi) [reply react-ui/src/components/Header.scss:8]: This is appreciated Seher! I remember this point of feedback during the previous Demo

File: react-ui/src/components/Header.js (hunk 4 above)
Comment (prdbondi) [reply react-ui/src/components/Header.js:9]: Appreciate the attention to detail to able to make the codebase current and concise!

**Review by prdbondi — APPROVED** [review]
Seher, I appreciate the work done on this PR! It goes a lng way to amking sure our front-end is properly tested! 
I have followed testing steps 1 and 2, and found no issues. On step 1, I was able to start the server and react-ui successfully, and on step 2 I observed the much-improved testing results to the level of coverage required! See screenshots below 


image


image

**Comment by sehersarin** [comment]
Thank you for the review, Peter! Glad to see that the code is working as expected :)

**Pull request merged**